pkg/endpoint/policy.go

```diff
@@ -160,43 +160,44 @@ func (e *Endpoint) sweepFilters(oldPolicy *policy.L4Policy,
 // Returns a map that represents all policies that were attempted to be removed;
 // it maps to whether they were removed successfully (true or false)
 func (e *Endpoint) removeOldFilter(identities *identityPkg.IdentityCache,
-	filter *policy.L4Filter) policy.SecurityIDContexts {
+	filter *policy.L4Filter, direction policymap.TrafficDirection) policy.SecurityIDContexts {
 
-	fromEndpointsSrcIDs := policy.NewSecurityIDContexts()
+	attemptedRemovedMapEntries := policy.NewSecurityIDContexts()
 	port := uint16(filter.Port)
 	proto := uint8(filter.U8Proto)
 
 	for _, sel := range getL4FilterEndpointSelector(filter) {
 		for _, id := range getSecurityIdentities(identities, &sel) {
 			srcID := id.Uint32()
 			l4RuleCtx, l7RuleCtx := e.ParseL4Filter(filter)
-			if _, ok := fromEndpointsSrcIDs[id]; !ok {
-				fromEndpointsSrcIDs[id] = policy.NewL4RuleContexts()
+			if _, ok := attemptedRemovedMapEntries[id]; !ok {
+				attemptedRemovedMapEntries[id] = policy.NewL4RuleContexts()
 			}
-			if err := e.PolicyMap.DeleteL4(srcID, port, proto, policymap.Ingress); err != nil {
+			if err := e.PolicyMap.DeleteL4(srcID, port, proto, direction); err != nil {
 				// This happens when the policy would add
 				// multiple copies of the same L4 policy. Only
 				// one of them is actually added, but we'll
 				// still try to remove it multiple times.
-				e.getLogger().WithError(err).WithField(logfields.L4PolicyID, srcID).Debug("Delete old l4 policy failed")
+				e.getLogger().WithError(err).WithFields(logrus.Fields{logfields.L4PolicyID: srcID,
+					logfields.TrafficDirection: direction}).Debug("deletion of old L4 policy failed")
```

Comment:
Could you reformat this so the fields are more clear? I kept wondering why you replaced WithField with WithFields until I noticed the additional field.
```golang
e.getLogger().WithError(err).WithFields(logrus.Fields{
    logfields.L4PolicyID:       srcID,
    logfields.TrafficDirection: direction,
}).Debug("deletion of old L4 policy failed")
```